subsys/net/ip/ipv6.c

```diff
@@ -202,8 +202,11 @@ static struct net_nbr *nbr_lookup(struct net_nbr_table *table,
 			continue;
 		}
 
-		if (nbr->iface == iface &&
-		    net_ipv6_addr_cmp(&net_ipv6_nbr_data(nbr)->addr, addr)) {
+		if (iface && nbr->iface != iface) {
```

Comment:
same question as the other patch, is it possible at this stage to have an nbr->iface which is null?

Reply:
Yes, it is possible now. That is why I needed to add this check.